modules/genomic_browser/ajax/genomic_file_upload.php

```diff
@@ -325,27 +365,40 @@ function (&$item, $key, $prefix) {
 
         $sample_count = count($headers);
 
-        while (($line = fgets($f)) !== false) {
-            $stmt = '';
-
-            $values   = explode(',', $line);
-            $values   = array_map('trim', $values);
-            $probe_id = $values[0];
-            // TODO validate probe value
-
-            array_shift($values);
-            $rows = array();
-            foreach ($values as $key => $value) {
-                $row = "('$headers[$key]', '$probe_id', $value)";
-                array_push($rows, $row);
-            }
-
-            $stmt .= join(',', $rows);
+        $stmt = $DB->prepare(
```

Comment:
The transaction handles that.
https://stackoverflow.com/questions/1415602/performance-in-pdo-php-mysql-transaction-versus-direct-execution